src/dmd/root/outbuffer.d

```diff
@@ -425,6 +430,45 @@ struct OutBuffer
     }
 }
 
+/**
+Utility to turn a slice into a length/ptr couple
+for printf-like interfaces
+
+This would be better done with some template trickery but
+the CTFE engine really wants to evaluate the slice at CT,
+and we just cannot accept that kind of attitude here,
+so we use string mixins to do the length/ptr separation
+and the cast on the length.
+
+Note that thanks to the way string mixings work,
+a caller must use `mixin("printf(format, " ~ mapSlices!Args() ~ ");");`,
+and not the more natural `printf(format, mixin(mapSlices!Args()));`
+
+Params:
+  Args = Type of the arguments to match
+
+Returns:
+  A string describing the parameters to pass to `printf`.
+  The parameters must be a tuple named `args`.
+*/
+string mapSlices(Args...)()
```

Comment:
Perhaps this is better located in `dmd.utils`.

Reply:
I was wondering that too, the issue I have is that `outbuffer` is supposed to be close to a leaf module AFAIK (it does import `rmem` and `rootobject`, but those are leaf).
Since utils is a grab-bag of dependency, I'm not so inclined to put it here.
Additionally, this function is very specific to `printf`.
But we could find a new home for it, as it doesn't feel like it belong to `outbuffer` so well either.